api/types/types.go

```diff
@@ -443,14 +443,8 @@ type EndpointResource struct {
 
 // NetworkCreate is the expected body of the "create network" http request message
 type NetworkCreate struct {
-	// Check for networks with duplicate names.
-	// Network is primarily keyed based on a random ID and not on the name.
-	// Network name is strictly a user-friendly alias to the network
-	// which is uniquely identified using ID.
-	// And there is no guaranteed way to check for duplicates.
-	// Option CheckDuplicate is there to provide a best effort checking of any networks
-	// which has the same name but it is not guaranteed to catch all name collisions.
-	CheckDuplicate bool
+	// Deprecated: CheckDuplicate is now always enabled.
+	CheckDuplicate bool `json:",omitempty"`
```

Comment:
See my other comment. I should've mentioned that in my original comments (sorry!) perhaps mention the API version in the deprecation message, and explain that it's still set by the _Client_ when connecting to older API versions;
https://github.com/moby/moby/blob/f8a4b6755d62fb3fdcfe6d3adff5756123080c37/api/types/system/info.go#L89